Added test cases for determinePosition, printBoard, and printError

smadhu247 opened the pull request:
Made these changes/ test cases

smadhu247 committed:
Added test cases for determinePosition, printBoard, and printError

bhagyesh-p commented on TicTacToe/src/com/company/BoardTest.java:
```diff
@@ -0,0 +1,64 @@
+        assertFalse(board.row2 == board.sec1 + "|" + board.sec2 + "|" + board.sec3);
+    }
+
+    @org.junit.jupiter.api.Test
```
remove this test, as of now it tests nothing

bhagyesh-p commented on TicTacToe/src/com/company/BoardTest.java:
```diff
@@ -0,0 +1,64 @@
+    Board board = new Board();
+    Main m = new Main();
+    @org.junit.jupiter.api.Test
+    void determinePosition() {
```
Add a case where we hit the return false statement/line

bhagyesh-p commented on TicTacToe/src/com/company/BoardTest.java:
```diff
@@ -0,0 +1,64 @@
+        assertEquals(7, board.determinePosition(3,1));
+        assertFalse(board.determinePosition(2,3) == 5);
+        assertEquals(8, board.determinePosition(3,2));
+        assertFalse(board.determinePosition(3,2) == 6);
+        assertEquals(9, board.determinePosition(3,3));
+        assertFalse(board.determinePosition(3,3) == 4);
+
```
Add a case where determinePosition returns -1, test all/as many cases.

Commit:
Changed 2 methods from void to String return type as a TEMPORARY testing measure.

Commit:
Replaced all comments OUTSIDE the methods in the Board class to be DocStrings to make it more readable

Changed the means of validation in method validatePosition by utilizing the .contains method in the ArrayList class.

Commit:
added DocStrings in Main.java

Commit:
modified the means of returning in printBoard and printError

Commit:
Moved the modified test methods into print statements so they continue to print their respective messages.

Changed the validatePositions method to only be 1 line

smadhu247 committed:
Added changes requested by the comments to printBoard test and printError test
